tabular/src/autogluon/tabular/models/vowpalwabbit/vowpalwabbit_model.py

```diff
@@ -0,0 +1,288 @@
+from autogluon.core.constants import BINARY, REGRESSION, MULTICLASS, SOFTCLASS, QUANTILE,\
+    PROBLEM_TYPES_CLASSIFICATION, PROBLEM_TYPES_REGRESSION
+from autogluon.core.constants import AG_ARGS_FIT
+from .vowpalwabbit_utils import VWFeaturesConverter
+
+
+class VowpalWabbitModel(AbstractModel):
+    """
+    VowpalWabbit Model: https://vowpalwabbit.org/
+
+    VowpalWabbit Command Line args: https://github.com/VowpalWabbit/vowpal_wabbit/wiki/Command-line-arguments
+
+    """
+    model_internals_file_name = 'model-internals.pkl'
+    namespace_separator_key = 'namespace_separator'
+    use_different_namespace_key = 'use_different_namespace'
+    UNSUPPORTED_PROBLEM_TYPES = [SOFTCLASS, QUANTILE]
+    sparse_weights = 'sparse_weights'
+
+    # Ref: https://github.com/VowpalWabbit/vowpal_wabbit/wiki/Loss-functions
+    CLASSIFICATION_LOSS_FUNCTIONS = ['logistic', 'hinge']
+    REGRESSION_LOSS_FUNCTIONS = ['squared', 'quantile', 'poisson', 'classic']
+
+    def __init__(self, **kwargs):
+        # Simply pass along kwargs to parent, and init our internal `_feature_generator` variable to None
+        super().__init__(**kwargs)
+        self.labels = None
+        self.oaa = None
+        self._load_model = None  # Used for saving and loading internal model file
+
+    # The `_preprocess` method takes the input data and transforms it to the internal representation usable by the model.
+    # `_preprocess` is called by `preprocess` and is used during model fit and model inference.
+    def _preprocess(self, X: pd.DataFrame, is_train=False, **kwargs) -> pd.Series:
+        print(f'Entering the `_preprocess` method: {len(X)} rows of data (is_train={is_train})')
+        X = super()._preprocess(X, **kwargs)
+        namespace_separator = kwargs.get(self.namespace_separator_key, '')
+        # self._feature_metadata contains the information related to features metadata.
+        X_series = VWFeaturesConverter(
+            namespace_separator=namespace_separator).convert_features_to_vw_format(X,
+                                                                                   self._feature_metadata.to_dict())
+        return X_series
+
+    # The `_fit` method takes the input training data (and optionally the validation data) and trains the model.
+    def _fit(self,
+             X: pd.DataFrame,  # training data
+             y: pd.Series,  # training labels
+             X_val=None,
+             y_val=None,
+             **kwargs):  # kwargs includes many other potential inputs, refer to AbstractModel documentation for details
+        print('Entering the `_fit` method')
+
+        try_import_vowpalwabbit()
+        from vowpalwabbit import pyvw
+
+        # Valid self.problem_type values include ['binary', 'multiclass', 'regression', 'quantile', 'softclass']
+        if self.problem_type in self.UNSUPPORTED_PROBLEM_TYPES:
+            raise TypeError(f"Vowpal Wabbit does not support {self.problem_type}")
```

Comment:
Define supported problem types, not unsupported. This stops working if a new problem type is added.